app/js/components/visitsAndEncounters.jsx

```diff
@@ -151,34 +152,51 @@ export class VisitsAndEncounters extends React.Component {
         output.push(data)
         data = []
       })
+    });
+    this.state.encountersWithNoVisits.map((encounter) => {
+      data["location"] = encounter.location.display
+      data["date"] = encounter.encounterDatetime
+      data["encounter"] = encounter.display
+      data["encounterUuid"] = encounter.uuid
+      output.push(data)
+      data = []
     })
-    return output
-  }
 
-  render() {
-    const columns = [{
-        id: 'type',
-        Header: 'Type',
-        accessor: rowProps => <a>{rowProps.encounter.slice(0, -10)}</a>,
-      },
-      {
-        id: 'location',
-        Header: 'Location',
-        accessor: rowProps => rowProps.location,
-      },
-      {
-        id: 'date',
-        Header: 'Date',
-        accessor: rowProps => rowProps.date.split("T")[0],
-      },
-      {
-        id: 'visit',
-        Header: 'Visit',
-        accessor: rowProps => <a>{rowProps.visit}</a>,
-      }
-    ];
-    return(
-      <div>
+    function compareEncountersByDate(firstEncounter, secondEncounter) {
+      if(firstEncounter.date > secondEncounter.date)
+        return -1;
```

Comment:
Looking at the general comparison routines, shouldn't 1 be returns for greater and then -1 for less than?

Reply:
Doesn't that depend on whether you are comparing for ascending or descending order? In this case i wanted a descending order to be consistent with the required reverse chronology. @dkayiwa

Reply:
The method body should never care about whether it is descending on ascending. That is up to the caller. Not so?
Have you looked at any compareTo method in the Java API to see when 1 and -1 are returned?

Reply:
Let me check it out